src/components/PivotTable/PivotTableBoard/useOnKeyNav.ts

```diff
@@ -0,0 +1,47 @@
+import { useCallback } from 'react'
+import { KeyNavigationDirection } from '../Droppable/types/model'
+
+interface UseOnKeyNavProps<T> {
+  setColumnKeys: (keys: Array<keyof T>) => void
+  setRowKeys: (keys: Array<keyof T>) => void
+  setAvailableKeys: (keys: Array<keyof T>) => void
+  columnKeys: Array<keyof T>
+  rowKeys: Array<keyof T>
+  availableKeys: Array<keyof T>
+}
+
+export const useOnKeyNav = <T extends object>(props: UseOnKeyNavProps<T>) => {
+  const { setColumnKeys, setRowKeys, setAvailableKeys, columnKeys, rowKeys, availableKeys } = props
+  const onKeyNav = useCallback(
+    (direction: KeyNavigationDirection, origin: string, key?: keyof T): boolean => {
+      const actions: Record<string, Partial<Record<KeyNavigationDirection, () => void>>> = {
+        filter: {
+          right: () => setColumnKeys([...columnKeys, key]),
+          down: () => setRowKeys([...rowKeys, key]),
+        },
+        column: {
+          right: () => setRowKeys([...rowKeys, key]),
+          down: () => setRowKeys([...rowKeys, key]),
+          left: () => setAvailableKeys([...availableKeys, key]),
+        },
+        row: {
+          right: () => setColumnKeys([...columnKeys, key]),
+          left: () => setAvailableKeys([...availableKeys, key]),
+          up: () => setAvailableKeys([...availableKeys, key]),
+        },
+      }
+
+      const action = actions[origin]?.[direction]
+
+      if (action) {
+        action()
+        return true
+      }
+
+      return false
```

Comment:
I don't like this type of using to `Record`, i think the `Record` should be used as a constant, but I'll ask for @Andrevmatias opinion, okay? It's a cleaner and smaller solution, but it's a wrong Record use for me

This is my suggestion:
```suggestion

  if (origin === "filter") {
    if (direction === "right") {
      setColumnKeys([...columnKeys, key])
      return true
    } else if (direction === "down") {
      setRowKeys([...rowKeys, key])
      return true
    }
   }
 // ...

return false
 }
```

This suggestion will be bigger than the actual solution :(

Reply:
I think the only mistake here is using a Record to express something that is actually an interface. Records should be used when you want all the keys to have a value (all the values of an array, for instance). Here I thinks it's better to create an interface like `KeyNavActions { filter: {...}, column: {...}, row: {...} }`.